src/Sarif.UnitTests/Writers/ResultLogJsonWriterTests.cs

```diff
@@ -86,13 +93,14 @@ public void ResultLogJsonWriter_RequiresNonNullIssue()
         {
             GetJson(uut =>
             {
-                uut.WriteToolAndRunInfo(s_defaultToolInfo, s_defaultRunInfo);
+                uut.WriteToolInfo(s_defaultToolInfo);
+                uut.WriteRunInfo(s_defaultRunInfo);
                 uut.WriteResult(null);
             });
         }
 
         [TestMethod]
-        [ExpectedException(typeof(ObjectDisposedException))]
+        [ExpectedException(typeof(InvalidOperationException))]
```

Comment:
Because we now filter all state checks (including asking the question 'are you already disposed?' through a common helper. rather than over-complicating the exception raising signature, I simply reduce all state confusion to InvalidOperationException